tests/functional/translation/test_adt_4.py

```diff
@@ -0,0 +1,13 @@
+from nagini_contracts.adt import ADT
+from typing import NamedTuple
+
+# 4 - Too many superclasses
+class ADT_4(ADT):
+    pass
+
+class AnySuperClass:
+    pass
+
+#:: ExpectedOutput(invalid.program:malformed.adt)
+class ADT_4_Cons(ADT_4, NamedTuple('ADT_4_Cons', []), AnySuperClass):
```

Comment:
Could you add translation tests where
- someone tries to declare a class whose superclass is an ADT constructor
- an ADT constructor class has a non-empty body (e.g., has methods)
- an ADT constructor class is declared whose superclass is directly the ADT class
- an ADT type without constructors is declared

Reply:
Added respectively:
- test_adt_7.py
- test_adt_8.py
- test_adt_5.py
- test_adt_9.py